### PR TITLE
EPMRPP-37395. Admin. Projects. Actions in menu.

### PR DESCRIPTION
**Pull request by ElenaRomanchuk**

### CODE REVIEW

File: app/src/controllers/user/actionCreators.js [hunk 1]
```diff
@@ -68,3 +70,40 @@ export const setStartTimeFormatAction = (format) => ({
...
+  const data = {
+    userNames: {
+      [userId]: userRole,
+    },
+  };
+  return fetch(URLS.userInviteInternal(projectName), {
+    method: 'put',
+    data,
+  }).then(() => {
+    dispatch({
```
Comment (margalencia) [inline app/src/controllers/user/actionCreators.js]: as far as I know we use sagas for async actions, not thunks, right @DiscoElevator ?

Comment (DiscoElevator) [reply app/src/controllers/user/actionCreators.js]: Event though we have thunks here, we plan to move to sagas, so I think we should start using them here at least for new actions.

File: app/src/controllers/administrate/projects/sagas.js [hunk 2]
```diff
@@ -37,6 +39,28 @@ function* watchSetViewMode() {
   yield takeEvery(START_SET_VIEW_MODE, setViewMode);
 }
 
+function* deleteProject({ payload: project }) {
+  try {
+    yield call(fetch, URLS.project(project.id), {
+      method: 'delete',
+    });
+  } catch (err) {
+    const error = (err.response && err.response.data && err.response.data.message) || err.message;
```
Comment (DiscoElevator) [inline app/src/controllers/administrate/projects/sagas.js]: I think it's always `err.message`

File: app/src/controllers/user/actionCreators.js [hunk 3]
```diff
@@ -68,3 +72,38 @@ export const setStartTimeFormatAction = (format) => ({
...
+  type: UNASSIGN_FROM_PROJECT,
+  payload: project,
+});
+
+export const unassignFromProjectSuccessAction = (project) => ({
+  type: UNASSIGN_FROM_PROJECT_SUCCESS,
+  payload: project,
+});
+
+// export const unassignFromProjectAction = (projectName, userId) => (dispatch) => {
```
Comment (DiscoElevator) [inline app/src/controllers/user/actionCreators.js]: Please delete commented code

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+      payload: { projectId: projectName, projectSection: MEMBERS },
+    });
+
+  onSettings = (projectName) =>
+    this.props.redirect({
+      type: PROJECT_DETAILS_PAGE,
+      payload: { projectId: projectName, projectSection: SETTINGS },
+    });
+
+  getMenuItems() {
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: `getMenuItems = () => ...`

File: app/src/controllers/user/sagas.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      }),
+    );
+    yield put(
+      showNotification({
+        messageId: 'assignSuccess',
+        type: NOTIFICATION_TYPES.SUCCESS,
+      }),
+    );
+  } catch (err) {
+    const error = (err.response && err.response.data && err.response.data.message) || err.message;
```
Comment (AmsterGet) [inline app/src/controllers/user/sagas.js]: See the https://github.com/reportportal/service-ui/pull/1310#discussion_r270774479

File: app/src/controllers/user/sagas.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    });
+    yield put(unassignFromProjectSuccessAction(project));
+    yield put(
+      showNotification({
+        messageId: 'unassignSuccess',
+        type: NOTIFICATION_TYPES.SUCCESS,
+      }),
+    );
+  } catch (err) {
+    const error = (err.response && err.response.data && err.response.data.message) || err.message;
```
Comment (AmsterGet) [inline app/src/controllers/user/sagas.js]: See the https://github.com/reportportal/service-ui/pull/1310#discussion_r270774479

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+      type: PROJECT_DETAILS_PAGE,
+      payload: { projectId: projectName, projectSection: SETTINGS },
+    });
+
+  getMenuItems = () => {
+    const { intl, project, isAssigned, userId } = this.props;
+    const isPersonalProject = project.projectName === `${userId}_personal`;
+    return [
+      {
+        onClick: () => this.onMembers(project.projectName),
```
Comment (AmsterGet) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: You are already can get this data directly from method

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+        label: intl.formatMessage(messages.members),
+        value: 'action-members',
+      },
+      {
+        onClick: () => this.onSettings(project.projectName),
```
Comment (AmsterGet) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: You are already can get this data directly from method

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+        label: intl.formatMessage(messages.settings),
+        value: 'action-settings',
+      },
+      {
+        type: SEPARATOR_ITEM,
+      },
+      {
+        onClick: () => this.onUnassign(project),
```
Comment (AmsterGet) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: You are already can get this data directly from method

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+        label: intl.formatMessage(messages.unassign),
+        value: 'action-unassign',
+        hidden: !isAssigned,
+        disabled: isPersonalProject,
+        title: isPersonalProject ? intl.formatMessage(messages.unassignFromPersonal) : '',
+      },
+      {
+        onClick: () => this.onAssign(project),
```
Comment (AmsterGet) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: You are already can get this data directly from method

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+        label: intl.formatMessage(messages.assign),
+        value: 'action-assign',
+        hidden: isAssigned,
+      },
+      {
+        onClick: () => this.onDelete(project),
```
Comment (AmsterGet) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: You are already can get this data directly from method

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+        },
+      },
+    });
+  };
+
+  onDeleteSubmit = ([project]) => {
+    this.props.deleteProject(project);
+  };
+
+  onMembers = (projectName) =>
```
Comment (AmsterGet) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: Unclear method name. Can it be f.e. `redirectToMembers`?

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+    this.props.redirect({
+      type: PROJECT_DETAILS_PAGE,
+      payload: { projectId: projectName, projectSection: MEMBERS },
+    });
+
+  onSettings = (projectName) =>
```
Comment (AmsterGet) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: The same here. Can it be f.e. `redirectToSettings`?

File: app/src/controllers/administrate/projects/reducer.test.js [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+      const payload = TABLE_VIEW;
+      const newState = setViewModeReducer(VIEW_MODE_INITIAL_STATE, {
+        type: SET_PROJECTS_VIEW_MODE,
+        payload,
+      });
+      expect(newState).toEqual(payload);
+    });
+  });
+
+  describe('projectFetchReducer', () => {
```
Comment (DiscoElevator) [inline app/src/controllers/administrate/projects/reducer.test.js]: Just a little tip: `fetchReducer` is already tested, so no need to write duplicating tests for every usage

**Pull request merged**